torch/ao/sparsity/experimental/pruner/base_pruner.py

```diff
@@ -44,45 +51,170 @@ def _prepare(self, use_path=False, *args, **kwargs):
+                    if isinstance(module, tuple(SUPPORTED_MODULES)):
+                        self.activation_handles.append(module.register_forward_hook(
+                            ActivationReconstruction(module.parametrizations.weight[0])
+                        ))
+                    else:
+                        raise NotImplementedError("This module type is not supported yet.")
+
+                else:  # needs zeros
+                    if getattr(module, 'mask', None) is None:
+                        module.register_buffer('mask', torch.tensor(module.weight.shape[0]))
+                    param = config.get('parametrization', ZeroesParametrization)
+                    parametrize.register_parametrization(module, 'weight', param(module.mask), unsafe=True)
+
+                if module.bias is not None:
+                    module.register_parameter('_bias', nn.Parameter(module.bias.detach()))
+                    module.bias = None
+                self.bias_handles.append(module.register_forward_hook(BiasHook(module.parametrizations.weight[0], self.prune_bias)))
+
+            if len(modules) == 2:  # (Conv2d, BN)
+                # should have the same set of pruned outputs
+                modules[1].parametrizations.weight[0].pruned_outputs = modules[0].parametrizations.weight[0].pruned_outputs
+
+
+    def prepare(self, model, config):
+        r"""Prepares a model, by adding the parametrizations and forward post-hooks.
+        Note::
+            The model is modified inplace. If you need to preserve the original
+            model, use copy.deepcopy.
+
+        Args:
+        - model [nn.Module]: model to configure. The model itself is not saved
+            but used for the state_dict saving / loading.
+        - config [list]: configuration elements could either be instances of
+            nn.Module or dict maps. The dicts must have a key 'module' with the
+            value being an instance of a nn.Module.
+        """
+        self.model = model  # TODO: Need to figure out how to load without this.
+        self.config = config
+
+        # If no config -- try getting all the supported layers
+        if self.config is None:
+            # Add all models to the config
+            self.config = []
+            stack = [model]
+            while stack:
+                module = stack.pop()
+                for name, child in module.named_children():
+                    if type(child) in SUPPORTED_MODULES:
+                        self.config.append(child)
+                    else:
+                        if type(child) in NEEDS_ZEROS and self.prune_bias:
+                            warnings.warn(f"Models with {type(child)} layers have config provided by user.")
+                        stack.append(child)
+
+        for module_config in self.config:
+            if type(module_config) is tuple:
+                first_layer, next_layer = module_config
```

Comment:
nit, would be good to assert that length is two and throw a good error message if it is not two